pkg/scheduler/plugins/coescheduling/core/gang_helper.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling/v1alpha1"
+
+	"encoding/json"
+
+	"github.com/koordinator-sh/koordinator/apis/extension"
+)
+
+func GetNamespaceSplicingName(namespace, name string) string {
+	return namespace + "/" + name
+}
+
+func GetGangNameByPod(pod *v1.Pod) string {
+	if pod == nil {
+		return ""
+	}
+	var gangName string
+	gangName = pod.Labels[v1alpha1.PodGroupLabel]
+	if gangName == "" {
+		gangName = pod.Annotations[extension.AnnotationGangName]
+	}
+	return gangName
+}
+
+func PgFromAnnotation(pod *v1.Pod) bool {
+	gangName := pod.Annotations[extension.AnnotationGangName]
+	return !(gangName == "")
```

Comment:
!(gangName == "") -> gangName != "";  besides GetGangNameByPod and PgFromAnnotation are different in parsing whether from pg crd or pod's annotation. the rule should be same.

Reply:
right, change PgFromAnnotation to isGangFromAnnotation, only check whether gang is from annotation